### PR TITLE
automated builds publishing to GCS bucket for sandbox, staging

### PR DESCRIPTION
**Pull request by critzo**
This PR adds support for automated Travis builds and publishing to GCS storage buckets. Sandbox builds have been tested and confirmed working. This PR does have a bunch of commits due to the extent of testing this new feature.

* Branches matching the pattern `sandbox-*` will be published to gs://website.mlab-sandbox.measurementlab.net/ , resolving to http://website.mlab-staging.measurementlab.net
* Merged PRs into `master` will be published to gs://website.mlab-staging.measurementlab.net/ resolving to https://website.mlab-staging.measurementlab.net/

When this PR is merged, the Travis build and push to the Staging bucket will be tested. Any build errors will be the result of incorrect permissions on the service account. If this happens, we'll readjust the permissions and rerun the build in Travis.

@georgiamoon Could you PTAL?

---
This change is [Reviewable](https://reviewable.io/reviews/m-lab/website/449)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -16,8 +16,9 @@ coverage
 _config-deploy.yml
 _config-deploy.ymle
 js/app.js
-_cf_s3_invalidator.yml
 .vagrant
 ubuntu-xenial-16.04-cloudimg-console.log
 .jekyll-metadata
-
+_cf_s3_invalidator.yml
```
Comment (georgiamoon) [inline .gitignore]: does the ordering here matter? it looks like this line was removed and then re-added later?

File: _pages/tests/bismark.md [hunk 2]
```diff
@@ -7,7 +7,7 @@ breadcrumb: tests
 
 # BISmark
 
-BISmark is an [OpenWRT](http://openwrt.org){:target="_blank"}-based platform that measures Internet service provider (ISP) performance and traffic inside home networks. BISmark takes the place of your home router, performing all of your router's normal functions in addition to periodic network performance measurements. BISmark users gain access to a personalized dashboard showing performance over time and comparing performance across ISPs and regions.
+BISmark is an OpenWRT-based (http://openwrt.org) platform that measures Internet service provider (ISP) performance and traffic inside home networks. BISmark takes the place of your home router, performing all of your router's normal functions in addition to periodic network performance measurements. BISmark users gain access to a personalized dashboard showing performance over time and comparing performance across ISPs and regions.
```
Comment (georgiamoon) [inline _pages/tests/bismark.md]: Any reason to remove this link? Also this seems like an unrelated change.

File: .gitignore (hunk 1 above)
Comment (critzo) [reply .gitignore]: This was a blank line being removed and added the AWS yaml file to .gitignore to keep it from being committed.

File: _pages/tests/bismark.md (hunk 2 above)
Comment (critzo) [reply _pages/tests/bismark.md]: I removed it because the URL didn't resolve and caused the build to fail. It might be back up by now, so we could re-add the link later. I left the URL string there as an option.

**Comment by critzo** [comment]
___
*[.travis.yml, line 8 at r1](https://reviewable.io/reviews/m-lab/website/449#-LXPj-4REykKiXUK7fQw:-LXPzizN4XWhVit-Aafb:b-kdmtie) ([raw file](https://github.com/m-lab/website/blob/02f9c729c06d01c733aff968dd196305501d235d/.travis.yml#L8)):*
<details><summary><i>Previously, stephen-soltesz (Stephen Soltesz) wrote…</i></summary><blockquote>

While this pattern is still present in many travis files, based on recent experience with mlab-ns, I believe it's best practice to not cache the google-cloud-sdk b/c it never seems to time-out and be replaced, which can lead to very old versions.
</blockquote></details>

ack'd. I'll remove that.

**Comment by critzo** [comment]
___
*[.travis.yml, line 45 at r1](https://reviewable.io/reviews/m-lab/website/449#-LXPjYMgBhuWUEEBN8Jy:-LXQ-82g2C835k2HK6v9:b-dw7623) ([raw file](https://github.com/m-lab/website/blob/02f9c729c06d01c733aff968dd196305501d235d/.travis.yml#L45)):*
<details><summary><i>Previously, stephen-soltesz (Stephen Soltesz) wrote…</i></summary><blockquote>

A common pattern in other .travis deploys is to order the sections from lowest to highest priority: sandbox, staging, then oti finally.
</blockquote></details>

ack'd. I'll reverse the current order to match this pattern.

**Comment by critzo** [comment]
___
*[.travis.yml, line 53 at r1](https://reviewable.io/reviews/m-lab/website/449#-LXPjoTV-6rd0Dzm5RGF:-LXQ-GEQ0Gw0EpLNPjTb:b-vsi32a) ([raw file](https://github.com/m-lab/website/blob/02f9c729c06d01c733aff968dd196305501d235d/.travis.yml#L53)):*
<details><summary><i>Previously, stephen-soltesz (Stephen Soltesz) wrote…</i></summary><blockquote>

What is this? Will this still be used once we can deploy to mlab-oti?
</blockquote></details>

the `after_deploy` step is residual from the AWS production setting. I'll remove it in my next commit.
